### DISCUSSION TITLE
[HW4] Definition of "Top down" in the homework

### DISCUSSION THREAD

**Discussion by JustinKim98** [prosyslab-classroom/cs424-program-reasoning#439] [Q&A]
### Your Name

Jaewoo Kim 김재우

### Question

숙제에 보면 Bottom up 및 Top down approach 2가지를 구현하는 것으로 되어 있습니다만
Top down을 구현할 때 성능을 높이기 위하여 Top down에서 Bottom up 과 비슷한 방법을 함께 활용하는 식으로 작성한다면 문제가 될까요? (Top down approach에서 크게 변하지 않는다는 가정 하에서)

If we read the homework, we must implement both bottom up and top down approaches.
To improve the performance, if one attmpts to combine the two, is it ok to call it "top down" approach if it's base implementation follows the top-down approach?

**Comment by KihongHeo** [comment]
#440 답변을 참고하세요.
